tests/unittests/Foundation/NSDictionaryTests.mm

```diff
@@ -68,3 +68,51 @@
 
     ASSERT_EQ(expectedCount, [dictionary count]);
 }
+
+TEST(NSDictionary, KeysSortedByValue) {
+    NSDictionary* dictionary = @{@"f" : @6, @"b": @2, @"a" : @1, @"c" : @3, @"e" : @5, @"d" : @4};
+    NSArray* expected = @[@"a", @"b", @"c", @"d", @"e", @"f"];
+    NSArray* actual = [dictionary keysSortedByValueUsingComparator:^NSComparisonResult(id obj1, id obj2) {
+        int a = [obj1 intValue];
+        int b = [obj2 intValue];
+        if (a == b) {
+            return NSOrderedSame;
+        }
+
+        return (a > b) ? NSOrderedDescending : NSOrderedAscending;
+    }];
+
+    ASSERT_OBJCEQ(expected, actual);
+}
+
+TEST(NSDictionary, KeysSortedByValueWithOptions) {
+    NSDictionary* dictionary = @{@"f" : @6, @"b": @2, @"a" : @1, @"c" : @3, @"e" : @5, @"d" : @4};
+    NSArray* expected = @[@"a", @"b", @"c", @"d", @"e", @"f"];
+    NSArray* actual = [dictionary keysSortedByValueWithOptions:0 usingComparator:^NSComparisonResult(id obj1, id obj2) {
+        int a = [obj1 intValue];
+        int b = [obj2 intValue];
+        if (a == b) {
+            return NSOrderedSame;
+        }
+
+        return (a > b) ? NSOrderedDescending : NSOrderedAscending;
+    }];
+
+    ASSERT_OBJCEQ(expected, actual);
+}
+
+TEST(NSDictionary, KeysSortedByValueWithOptions_Stable) {
+    NSDictionary* dictionary = @{@"a" : @1, @"b": @1, @"c" : @1, @"d" : @1, @"e" : @1, @"f" : @0};
+    NSArray* expected = @[@"f", @"d", @"b", @"e", @"c", @"a"]; // Note: ordering after "f" is dependent on CFDictionary (this ordering matches the reference platform)
+    NSArray* actual = [dictionary keysSortedByValueWithOptions:NSSortStable usingComparator:^NSComparisonResult(id obj1, id obj2) {
```

Comment:
Dustin's point is that stable sort only makes sense if there are two items that are being sorted are of same value - then they are expected in the same order.  As is, this probably will just pick a different sort algorithm.

---
In reply to: [72361429](https://github.com/Microsoft/WinObjC/pull/673#discussion_r72361429) [](ancestors = 72361429,72359377)